x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { IScopedClusterClient } from 'kibana/server';
+import { JOB_MAP_NODE_TYPES } from '../../../common/constants/data_frame_analytics';
+import { INDEX_META_DATA_CREATED_BY } from '../../../common/constants/file_datavisualizer';
+import { getAnalysisType } from '../../../common/util/analytics_utils';
+import {
+  AnalyticsMapEdgeElement,
+  AnalyticsMapReturnType,
+  AnalyticsMapNodeElement,
+  isAnalyticsMapEdgeElement,
+  isAnalyticsMapNodeElement,
+  isIndexPatternLinkReturnType,
+  isJobDataLinkReturnType,
+  isTransformLinkReturnType,
+  MapElements,
+  NextLinkReturnType,
+} from './types';
+
+export class AnalyticsManager {
+  private _client: IScopedClusterClient['asInternalUser'];
+  public _inferenceModels: any; // TODO: update types
+
+  constructor(client: IScopedClusterClient['asInternalUser']) {
+    this._client = client;
+    this._inferenceModels = [];
+  }
+
+  public set inferenceModels(models: any) {
+    this._inferenceModels = models;
+  }
+
+  public get inferenceModels(): any {
+    return this._inferenceModels;
+  }
+
+  async setInferenceModels() {
+    try {
+      const models = await this.getAnalyticsModels();
+      this.inferenceModels = models;
+    } catch (error) {
+      // TODO: bubble up this error?
+      // eslint-disable-next-line
+      console.error('Unable to fetch inference models', error);
+    }
+  }
+
+  private isDuplicateElement(analyticsId: string, elements: any[]): boolean {
+    let isDuplicate = false;
+    elements.forEach((elem: any) => {
+      if (elem.data.label === analyticsId && elem.data.type === JOB_MAP_NODE_TYPES.ANALYTICS) {
+        isDuplicate = true;
+      }
+    });
+    return isDuplicate;
```

Comment:
check with `some` seems more suitable for this case 
```suggestion
    return elements.some((elem: any) => {
      return elem.data.label === analyticsId && elem.data.type === JOB_MAP_NODE_TYPES.ANALYTICS;
      }
    });
```